src/Sentry/TransactionTracer.cs

```diff
@@ -210,59 +207,75 @@ public TransactionTracer(IHub hub, ITransactionContext context)
         Status = context.Status;
         IsSampled = context.IsSampled;
+        StartTimestamp = _stopwatch.StartDateTimeOffset;
+
+        if (context is TransactionContext transactionContext)
+        {
+            _instrumenter = transactionContext.Instrumenter;
+        }
     }
 
     /// <inheritdoc />
-    public void AddBreadcrumb(Breadcrumb breadcrumb) =>
-        _breadcrumbs.Add(breadcrumb);
+    public void AddBreadcrumb(Breadcrumb breadcrumb) => _breadcrumbs.Add(breadcrumb);
 
     /// <inheritdoc />
-    public void SetExtra(string key, object? value) =>
-        _extra[key] = value;
+    public void SetExtra(string key, object? value) => _extra[key] = value;
 
     /// <inheritdoc />
-    public void SetTag(string key, string value) =>
-        _tags[key] = value;
+    public void SetTag(string key, string value) => _tags[key] = value;
 
     /// <inheritdoc />
-    public void UnsetTag(string key) =>
-        _tags.TryRemove(key, out _);
+    public void UnsetTag(string key) => _tags.TryRemove(key, out _);
 
     /// <inheritdoc />
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public void SetMeasurement(string name, Measurement measurement) =>
-        _measurements[name] = measurement;
+    public void SetMeasurement(string name, Measurement measurement) => _measurements[name] = measurement;
 
-    internal ISpan StartChild(SpanId parentSpanId, string operation)
-    {
-        // Limit spans to 1000
-        var isOutOfLimit = _spans.Count >= 1000;
+    /// <inheritdoc />
+    public ISpan StartChild(string operation) => StartChild(SpanId, operation);
 
-        var span = new SpanTracer(_hub, this, parentSpanId, TraceId, operation)
+    internal ISpan StartChild(SpanId parentSpanId, string operation, Instrumenter instrumenter = Instrumenter.Sentry) =>
+        StartChild(null, parentSpanId, operation, instrumenter);
+
+    internal ISpan StartChild(SpanId? spanId, SpanId parentSpanId, string operation,
+        Instrumenter instrumenter = Instrumenter.Sentry)
+    {
+        if (instrumenter != _instrumenter)
         {
-            IsSampled = !isOutOfLimit
-                ? IsSampled
-                : false // sample out out-of-limit spans
-        };
+            SentrySdk.CurrentOptions?.LogWarning(
+                $"Attempted to create a span via {instrumenter} instrumentation to a span or transaction" +
+                $" originating from {_instrumenter} instrumentation. The span will not be created.");
+            return NoOpSpan.Instance;
+        }
 
-        if (!isOutOfLimit)
+        var span = new SpanTracer(_hub, this, parentSpanId, TraceId, operation);
+        if (spanId is { } id)
         {
-            _spans.Add(span);
+            span.SpanId = id;
         }
 
+        AddChildSpan(span);
         return span;
     }
 
-    /// <inheritdoc />
-    public ISpan StartChild(string operation) =>
-        StartChild(SpanId, operation);
+    private void AddChildSpan(SpanTracer span)
```

Comment:
Is this logic also defined elsewhere? I recall we already had a 1k span limit within a transaction.

Please be aware we're investigating spans outside transactions so will be good to have this logic in a single place/easily acessibly if/when we change this in the future (no more limitation when capturing outside tranasactions).

Reply:
I can't find the logic anywhere else so I think we're good... it will die with the TransactionTracer eventually.